tests/docs/multi_dut_asic_testing_support.md

```diff
@@ -0,0 +1,499 @@
+# Improve the test infrastructure to support multi-DUT and multi-ASIC systems
+
+## Background
+
+PR https://github.com/Azure/SONiC/pull/644 introduced the HLD to support multi ASIC. In the future, multi DUT or Chassis will be supported by SONiC as well. The test infrastructure and some of the customized ansible modules need to be updated to support testing of the upcoming new architectures. This document tries to propose how to improve the current test infrastructure to support multi-DUT and multi-ASIC systems. The target is to ensure that the existing test scripts are not broken and we can update the tests in incremental way.
+
+## Current Status
+
+In current infrastructure, we use class `SonicHost` to represent a single SONiC system.
+
+Fixture `duthost` is an instance of `SonicHost`. Fixture `duthosts` is a list of `SonicHost` instances initialized according to the list of DUT host names defined in `testbed.csv` for current testbed.
+
+This `SonicHost` class provides interface for calling any ansible module on the SONiC device under test using format like `duthost.<module_name>(*args, **kwargs)`. Please refer to the implementation of methods [`__getattr__` in `AnsibleHostBase`](https://github.com/Azure/sonic-mgmt/blob/857cfb2bbed741057e830a1a20b48d2364f65cc3/tests/common/devices.py#L58) for more details. `AnsibleHostBase` is the base class of `SonicHost`.
+
+## Requirements
+
+To support multi-DUT and multi-ASIC testing, the `duthosts` and `duthost` fixture need to be updated. The target is that in test scripts we can use these fixtures to perform operations for
+* all DUTs
+* specific type of DUTs
+* specific DUT
+* all ASICs
+* specific ASIC
+* combinations of the above scenarios
+
```

Comment:
it would be great if you have have some diagram to show the relation between all classes you plan to add to represent the multi-asic, multi-dut.

Reply:
Sure, I'll update.